src/tests/account/test_eth_account.cairo

```diff
@@ -0,0 +1,613 @@
+    assert(supports_default_interface, 'Should support base interface');
+
+    let supports_account_interface = mock_state.supports_interface(ISRC6_ID);
+    assert(supports_account_interface, 'Should support account id');
+}
+
+#[test]
+fn test_assert_only_self_true() {
+    let mut state = COMPONENT_STATE();
+
+    testing::set_contract_address(ACCOUNT_ADDRESS());
+    testing::set_caller_address(ACCOUNT_ADDRESS());
+    state.assert_only_self();
+}
+
+#[test]
+#[should_panic(expected: ('EthAccount: unauthorized',))]
+fn test_assert_only_self_false() {
+    let mut state = COMPONENT_STATE();
+
+    testing::set_contract_address(ACCOUNT_ADDRESS());
+    let other = contract_address_const::<0x4567>();
+    testing::set_caller_address(other);
+    state.assert_only_self();
+}
+
+#[test]
+fn test__is_valid_signature() {
+    let mut state = COMPONENT_STATE();
+    let data = SIGNED_TX_DATA();
+    let hash = data.transaction_hash;
+
+    let mut bad_signature = data.signature;
+
+    bad_signature.r += 1;
+
+    let mut serialized_good_signature = array![];
+    let mut serialized_bad_signature = array![];
+
+    data.signature.serialize(ref serialized_good_signature);
+    bad_signature.serialize(ref serialized_bad_signature);
+
+    state.initializer(data.public_key);
+
+    let is_valid = state._is_valid_signature(hash, serialized_good_signature.span());
+    assert(is_valid, 'Should accept valid signature');
+
+    let is_valid = state._is_valid_signature(hash, serialized_bad_signature.span());
+    assert(!is_valid, 'Should reject invalid signature');
+}
+
+#[test]
+fn test__set_public_key() {
+    let mut state = COMPONENT_STATE();
+    let public_key = ETH_PUBKEY();
+    state._set_public_key(public_key);
+
+    assert_only_event_owner_added(ZERO(), public_key);
+
+    let public_key = state.get_public_key();
+    assert(public_key == public_key, 'Should update key');
+}
+
+//
+// Helpers
+//
+
+fn assert_event_owner_added(contract: ContractAddress, public_key: EthPublicKey) {
+    let event = utils::pop_log::<OwnerAdded>(contract).unwrap();
+    let guid = get_guid_from_public_key(public_key);
+    assert(event.new_owner_guid == guid, 'Invalid `new_owner_guid`');
+}
+
+fn assert_only_event_owner_added(contract: ContractAddress, public_key: EthPublicKey) {
+    assert_event_owner_added(contract, public_key);
+    utils::assert_no_events_left(contract);
+}
+
+fn assert_event_owner_removed(contract: ContractAddress, public_key: EthPublicKey) {
+    let event = utils::pop_log::<OwnerRemoved>(contract).unwrap();
+    let guid = get_guid_from_public_key(public_key);
+    assert(event.removed_owner_guid == guid, 'Invalid `removed_owner_guid`');
+}
+
+fn get_guid_from_public_key(public_key: EthPublicKey) -> felt252 {
+    let (x, y) = public_key.get_coordinates().unwrap();
+    poseidon_hash_span(array![x.low.into(), x.high.into(), y.low.into(), y.high.into()].span())
+}
```

Comment:
I don't think so, because then we wouldn't be testing against changes on that method, and a bug introduced would be translated to tests automatically.

Reply:
That's fair. We could also independently test `_get_guid_from_public_key`. That way, a buggy change would also raise a flag and be easily identified. No issue with the current approach though